src/tokens/Assets.tsx

```diff
@@ -0,0 +1,442 @@
+
+const DEVICE_WIDTH_BREAKPOINT = 340
+
+type Props = NativeStackScreenProps<StackParamList, Screens.Assets>
+interface SectionData {
+  appName?: string
+}
+
+const AnimatedSectionList =
+  Animated.createAnimatedComponent<SectionListProps<TokenBalance | Position, SectionData>>(
+    SectionList
+  )
+
+const assetIsPosition = (asset: Position | TokenBalance): asset is Position =>
+  'type' in asset && (asset.type === 'app-token' || asset.type === 'contract-position')
+
+export enum AssetTabType {
+  Tokens = 0,
+  Collectibles = 1,
+  Positions = 2,
+}
+
+// offset relative to the bottom of the non sticky header component, where the
+// screen header opacity animation starts
+const HEADER_OPACITY_ANIMATION_START_OFFSET = 44
+// distance in points over which the screen header opacity animation is applied
+const HEADER_OPACITY_ANIMATION_DISTANCE = 20
+
+function AssetsScreen({ navigation, route }: Props) {
+  const { t } = useTranslation()
+
+  const activeTab = route.params?.activeTab ?? AssetTabType.Tokens
+
+  const supportedNetworkIds = getSupportedNetworkIdsForTokenBalances()
+  const tokens = useTokensWithTokenBalance()
+  const localCurrencySymbol = useSelector(getLocalCurrencySymbol)
+  const totalTokenBalanceLocal = useTotalTokenBalance() ?? new BigNumber(0)
+  const tokensAreStale = useTokenPricesAreStale(supportedNetworkIds)
+
+  const insets = useSafeAreaInsets()
+
+  // TODO: Update this to filter out unsupported networks once positions support non-Celo chains
+  const positions = useSelector(positionsSelector)
+  const showPositions = getFeatureGate(StatsigFeatureGates.SHOW_POSITIONS)
+  const displayPositions = showPositions && positions.length > 0
+
+  const dappShortcutsEnabled = getFeatureGate(StatsigFeatureGates.SHOW_CLAIM_SHORTCUTS)
+  const positionsWithClaimableRewards = useSelector(positionsWithClaimableRewardsSelector)
+  const showClaimRewards =
+    dappShortcutsEnabled &&
+    positionsWithClaimableRewards.length > 0 &&
+    activeTab !== AssetTabType.Collectibles
+
+  // TODO: Update these to filter out unsupported networks once positions support non-Celo chains
+  const totalPositionsBalanceUsd = useSelector(totalPositionsBalanceUsdSelector)
+  const totalPositionsBalanceLocal = useDollarsToLocalAmount(totalPositionsBalanceUsd)
+  const totalBalanceLocal = totalTokenBalanceLocal?.plus(totalPositionsBalanceLocal ?? 0)
+
+  const [nonStickyHeaderHeight, setNonStickyHeaderHeight] = useState(0)
+  const [listHeaderHeight, setListHeaderHeight] = useState(0)
+  const [listFooterHeight, setListFooterHeight] = useState(0)
+
+  const scrollPosition = useSharedValue(0)
+  const footerPosition = useSharedValue(0)
+  const handleScroll = useAnimatedScrollHandler<{ prevScrollY: number }>(
+    {
+      onScroll: (event, ctx) => {
+        const scrollY = event.contentOffset.y
+        scrollPosition.value = scrollY
+
+        function clamp(value: number, min: number, max: number) {
```

Comment:
I think lodash [has a clamp method](https://www.geeksforgeeks.org/lodash-_-clamp-method/) built in.